src/OpenTelemetry.Instrumentation.Wcf/Implementation/TelemetryPropagationWriter.cs

```diff
@@ -83,7 +83,7 @@ public static void HttpRequestHeaders(Message request, string name, string value
     public static void Default(Message request, string name, string value)
     {
         SoapMessageHeaders(request, name, value);
-        if (WcfInstrumentationActivitySource.Options.SuppressDownstreamInstrumentation)
+        if (WcfInstrumentationActivitySource.Options?.SuppressDownstreamInstrumentation ?? false)
```

Comment:
Interesting here you are handling `null` options (`Options?`) but other places you expect it to be non-null (`Options!`). I don't have any issue with that but we _could_ be a bit more consistent. Nice thing about handling `null` is then I won't ask for `Debug.Assert`s 😄

Reply:
IIRC my thought was all the other places are actually impossible to physically get to unless this module has been initialized and is active (and thus Options must already be non-null)....this location originally was public and could possibly be called directly before Options was initialized (altho I have no idea why someone would do that) but yeah I don't think that's actually possible anymore.